src/utils/OrderedMsgChain.js

```diff
@@ -20,6 +20,7 @@ export default class OrderedMsgChain extends EventEmitter {
         propagationTimeout = DEFAULT_PROPAGATION_TIMEOUT, resendTimeout = DEFAULT_RESEND_TIMEOUT,
     ) {
         super()
+        this.markedExplicitly = new WeakSet()
```

Comment:
Possible this should be a regular `Set` that stores serialized `messageRefs` instead, in the case that a copy of the message is created and passed to the ordering util.